library/core/tests/ascii_char.rs

```diff
@@ -0,0 +1,68 @@
+use core::ascii::Char;
+use core::fmt::Write;
+
+/// Tests Display implementation for ascii::Char.
+#[test]
+fn test_display() {
+    let want = (0..128u8).map(|b| b as char).collect::<String>();
+    let mut got = String::with_capacity(128);
+    for byte in 0..128 {
+        write!(&mut got, "{}", Char::from_u8(byte).unwrap()).unwrap();
+    }
+    assert_eq!(want, got);
+}
+
+/// Tests Debug implementation for ascii::Char.
+#[test]
+fn test_debug() {
+    for (chr, want) in [
+        // Control characters
+        (Char::Null, r#"'\0'"#),
+        (Char::StartOfHeading, r#"'\x01'"#),
+        (Char::StartOfText, r#"'\x02'"#),
+        (Char::EndOfText, r#"'\x03'"#),
```

Comment:
Except for the `QuotationMark` case, these could be perhaps slightly more legible by dropping the hashes, like `r"'\0'"` instead of `r#"'\0'"#`.

Also I could come up with a straightforward test to relate/equate the outputs for `AsciiChar` with the one from `char`, as follows; perhaps you want to include something like that, too:

```rs
for n in 0..128_u8 {
    let debug_char = format!("{:?}", n as char);
    let debug_ascii = format!("{:?}", ascii::Char::from_u8(n).unwrap());
    if let Some(rest) = debug_char.strip_prefix(r"'\u{") {
        assert_eq!(
            debug_ascii,
            format!(r"'\x{:0>2}'", rest.strip_suffix("}'").unwrap())
        );
    } else {
        assert_eq!(debug_ascii, debug_char)
    }
}
```